packages/contracts/contracts/registry/Registry.sol

```diff
@@ -4,36 +4,49 @@
 
 pragma solidity 0.8.10;
 
-import "../core/DAO.sol";
+import "../core/component/Component.sol";
+
+/// @title Register a DAO
+/// @author Aragon Association - 2022
+/// @notice This contract allows to register a DAO by its address
+contract Registry is Component {
+    bytes32 public constant REGISTER_DAO_ROLE = keccak256("REGISTER_DAO_ROLE");
+
+    mapping(IDAO => bool) public daos;
+
+    bytes4 internal constant REGISTRY_INTERFACE_ID = this.register.selector;
 
-/// @title Register your unique DAO name
-/// @author Samuel Furter - Aragon Association - 2021
-/// @notice This contract provides the possiblity to register a DAO by a unique name.
-contract Registry {
     /// @notice Thrown if the DAO name is already registered
-    /// @param name The DAO name requested for registration
-    error RegistryNameAlreadyUsed(string name);
+    /// @param dao The address requested for registration
+    error DAOAlreadyRegistered(IDAO dao);
```

Comment:
Couldn't we just emit the DAO address instead of IDAO?
Or this is converted to address under the hood?

Reply:
> Or this is converted to address under the hood?

Yes, it becomes an address.